### PR TITLE
feat: include variable description as exported dataset as well as a function to get the required variables

### LINKED ISSUE #16: Function to list all registers needed as well as all variables needed
So that it is very explicit, in order for this algorithm to work, the user needs THESE registers and THESE variables.

### PR DESCRIPTION
**Pull request by lwjohnst86**
This closes #16 and is a stacked PR the depends on #45.

I originally had the `required_variables` as an internal dataset, but it makes more sense that more of the `variable-description` data is "exported" (available to the user) so that they can use that if needed. Converting into a "exported" object is done by saving it to `data/` as a `.rda` object (via `usethis::use_data()`).

I included documentation on that dataset in `R/variable-description.R`.

I created an `get_` helper function to extract the required variables from a given register (along with unit tests for the helper functions).

### CODE REVIEW

File: R/get-variables.R [hunk 1]
```diff
@@ -2,10 +2,31 @@
 #' Get a list of the registers' abbreviations.
 #'
 #' @return A character string.
-#' @export
+#' @keywords internal
```
Comment (signekb) [inline R/get-variables.R:5]: Ah, nice!

File: R/variable-description.R [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#' Variables from registers and their descriptions that are required for the
+#' OSDC algorithm.
```
Comment (signekb) [inline R/variable-description.R:2]: OSDC is capitalised here. Should we do this everywhere?

File: R/variable-description.R [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+#' \describe{
+#'   \item{register_name}{The official, full Danish name of the register.}
+#'   \item{register_abbrev}{The official abbreviation for the register.}
+#'   \item{variable_name}{The official name of the variable found in the register.}
+#'   \item{years_covered}{The years when the variable is available from.}
+#'   \item{danish_description}{The official description in Danish for the variable.}
+#'   \item{english_description}{The translated description in English for the variable.}
+#' }
+#' @source <https://www.dst.dk/extranet/forskningvariabellister/Oversigt%20over%20registre.html>
+"variable_description"
```
Comment (signekb) [inline R/variable-description.R:15]: I'm not sure I understand this. The html includes a table with 14 columns?

File: R/verify-variables.R [hunk 4]
```diff
@@ -1,8 +1,12 @@
 #' Verify that the dataset has the required variables for the algorithm.
 #'
+#' Use this function within an `if` condition inside a function to provide an
```
Comment (signekb) [inline R/verify-variables.R:3]:
```suggestion
#' Use this function inside another function within an `if` condition to provide an
```

File: R/variable-description.R (hunk 3 above)
Comment (lwjohnst86) [reply R/variable-description.R:17]: Good catch.

File: R/variable-description.R (hunk 2 above)
Comment (Aastedet) [reply R/variable-description.R]: It is an abbreviation, so I suppose we should when refering to the algorithm, but keep the package name as lower-case. That's how I did it in the validation paper, but I have no strong feelings about it.

Comment (signekb) [reply R/variable-description.R]: I'll resolve this based on #61

**Pull request merged**